test/org/javarosa/core/form/api/test/OutputInComputedConstraintTextTest.java

```diff
@@ -0,0 +1,86 @@
+package org.javarosa.core.form.api.test;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import org.javarosa.core.PathConst;
+import org.javarosa.core.model.FormDef;
+import org.javarosa.core.model.FormIndex;
+import org.javarosa.core.model.IFormElement;
+import org.javarosa.core.model.QuestionDef;
+import org.javarosa.core.model.data.LongData;
+import org.javarosa.core.services.PrototypeManager;
+import org.javarosa.core.test.FormParseInit;
+import org.javarosa.form.api.FormEntryCaption;
+import org.javarosa.form.api.FormEntryController;
+import org.javarosa.form.api.FormEntryModel;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.junit.Before;
+import org.junit.Test;
+
+public class OutputInComputedConstraintTextTest {
+  static {
+    PrototypeManager.registerPrototype("org.javarosa.model.xform.XPathReference");
+  }
+
+  private Map<String, FormIndex> formIndexesById = new HashMap<>();
+  private FormDef formDef;
+  private FormEntryController ctrl;
+  private FormEntryModel model;
+
+  @Before
+  public void setUp() {
+    FormParseInit fpi = new FormParseInit();
+    fpi.setFormToParse((new File(PathConst.getTestResourcePath(), "constraint-message-error.xml")).getAbsolutePath());
```

Comment:
We decided some months back that new resource files used by tests would be placed into a subdirectory matching the package name. `resources/org/javarosa/core/form/api/test/constraint-message-error.xml` in this case.

You can use `r` to specify this path automatically.

`import static org.javarosa.test.utils.ResourcePathHelper.r;`
`fpi.setFormToParse(r("constraint-message-error.xml").toString());`

Reply:
Alright! Do you think the test should be in that package?